test/kafkaesque/integration_test.exs

```diff
@@ -36,14 +60,14 @@ defmodule Kafkaesque.IntegrationTest do
     assert message.body == Jason.encode!(%{hello: :kafka})
     assert message.state == :pending
 
-    # Could perform some synchronization to avoid sleeping
-    :timer.sleep(1000)
+    # Await message acknowledgement
+    await_telemetry_event()
 
     message2 = Repo.reload(message)
-
     assert message2.state == :published
   end
 
+  @tag sync_on_telemetry_event: [:kafkaesque, :publish, :start]
```

Comment:
Maybe could hook into `[:kafkaesque, :produce, :stop]` instead. Leaves a bigger window for us. Not too big of a difference though since they are successive.

Reply:
Actually, I think its better as is :)